bench-tps/src/bench.rs

```diff
@@ -561,34 +609,52 @@ fn generate_nonced_system_txs<T: 'static + BenchTpsClient + Send + Sync + ?Sized
 ) -> Vec<TimestampedTransaction> {
     let length = source.len();
     let mut transactions: Vec<TimestampedTransaction> = Vec::with_capacity(length);
-    for i in 0..length {
-        let (from, to, nonce, nonce_blockhash) = if !reclaim {
-            (
-                source[i],
-                dest[i],
-                source_nonce[i],
-                get_nonce_blockhash(client.clone(), source_nonce[i].pubkey()),
-            )
-        } else {
-            (
-                dest[i],
-                source[i],
-                dest_nonce[i],
-                get_nonce_blockhash(client.clone(), dest_nonce[i].pubkey()),
-            )
-        };
+    if !reclaim {
+        let pubkeys: Vec<Pubkey> = source_nonce
+            .iter()
+            .map(|keypair| keypair.pubkey())
+            .collect();
 
-        transactions.push((
-            system_transaction::nonced_transfer(
-                from,
-                &to.pubkey(),
-                1,
-                &nonce.pubkey(),
-                from,
-                nonce_blockhash,
-            ),
-            None,
-        ));
+        let blockhashes: Vec<Hash> = pubkeys
+            .chunks(MAX_MULTIPLE_ACCOUNTS)
+            .map(|pubkeys| get_nonce_blockhashes(&client, &pubkeys))
+            .flatten()
+            .collect();
```

Comment:
nit: Instead of chunking here, it might be faster to pass all of them to `get_nonce_blockhashes` and then do the chunking in the function, that way you can always fetch `MAX_MULTIPLE_ACCOUNTS`.  What do you think?

Reply:
I've updated the code and tested it with private cluster.

I've added UPD section to the PR description with some plots from this run. 
It looks like tps is spiky because we request all the blockhashes and later execute all the transactions. Spike width is ~4s while the gap is ~2s => it is possible to increase tps at most by 30% by overlapping these two phases.

Reply:
Awesome, and yes it makes sense to do that work separately